streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala

```diff
@@ -636,7 +640,16 @@ class StreamingContext private[streaming] (
    *         if the waiting time elapsed before returning from the method.
    */
   def awaitTerminationOrTimeout(timeout: Long): Boolean = {
-    waiter.waitForStopOrError(timeout)
+    try {
+      waiter.waitForStopOrError(timeout)
+    } catch {
+      case e: InterruptedException =>
+        logger.info("StreamingContext interrupted at awaitTerminationOrTimeout")
+        false
```

Comment:
You can't return `false` in this case, and I also don't think there is value in also logging any exception here. Refer to the JIRA for the intended change here, which was just a documentation update. You might close this PR and just open a new one.